### PR TITLE
Require the isomorphic React instead of internals from renderers

### PR DESCRIPTION
**Pull request by sebmarkbage**
This stops using internals of the isomorphic package inside of renderers. This puts our renderers on equal footing with any other renderer built on top of the isomorphic package.

This is needed for flat builds. It also lets us get rid of a bunch of special cases in the build scripts.

It also allow us to just copy the source files into React Native instead of having to build first to resolve the special cases. Because React Native uses providesModule too and has an alias for `React` which uses the npm package.

### CODE REVIEW

**Comment by sebmarkbage** [comment]
cc @zpao @spicyj

**Comment by spicyj** [comment]
k

File: src/shared/utils/traverseAllChildren.js [hunk 1]
```diff
@@ -23,6 +23,12 @@ var SEPARATOR = '.';
 var SUBSEPARATOR = ':';
 
 /**
+ * This is inlined from ReactElement since this file is shared between
+ * isomorphic and renderers. We could extract this to a
```
Comment (gaearon) [inline src/shared/utils/traverseAllChildren.js]: To a what? 🙂

Comment (sebmarkbage) [inline src/shared/utils/traverseAllChildren.js]: ugh. Haha. I moved this down below instead of finishing this comment.

**Pull request merged**